pytest/lib/adversary.py

```diff
@@ -0,0 +1,35 @@
+import os, subprocess
+
+ADVERSARIAL_TEST = False
+
+HONEST_PATH = "../target/debug"
+ADVERSARIAL_PATH = "./.adversary/debug"
+
+def prepare_adversarial_binary():
+    honest_near = os.path.join(HONEST_PATH, "near")
+    adversarial_near = os.path.join(ADVERSARIAL_PATH, "near")
+    assert os.path.exists(honest_near)
+
+    need_to_recompile = False
+    if not os.path.exists(adversarial_near):
+        print("Adversarial binary is missing, recompiling")
+        need_to_recompile = True
+    else:
+        honest_mtime = os.path.getmtime(honest_near)
+        adversarial_mtime = os.path.getmtime(adversarial_near)
+
+        if honest_mtime > adversarial_mtime:
+            print("Adversarial binary is older than honest, recompiling")
+            need_to_recompile = True
```

Comment:
Why honest near is somehow relevant to adversarial?

Reply:
This is used to detect if there are new changes in the binary and we need to recompile adversarial binary.

As it is right now it doesn't work very nice since it is possible that there are new changes in the code, but the honest binary haven't been recompiled. I'll change and let cargo detect by itself if there are any new change.

Reply:
I guess it's much better. If nothing to recompile, cargo execution finishes almost instantly.